security/fma-withdrawals-root-in-block-header.md

```diff
@@ -0,0 +1,90 @@
+  
+  Triggers: 
+  
+  * A failed hardfork activation in the execution client.
+
+  * If there is an execution client bug, for example it is possible the root is (incorrectly) added to the header before the state is fully committed. 
+
+  * If we were to ever introduce non empty `withdrawals` in the block body, this might override the mechanism introduced with this feature and invalidate the interpreation of the `withdrawalsRoot` field. 
+
+- **Risk Assessment:**
+
+  **High impact, low likelihood** 
+  
+  Temporary downtime for withdrawals, or loss of funds if not remediated in time to challenge a malicious proposal.
+
+  **Mitigations:**
+  * We rely on e2e tests to check for consistency between the outputs returned from op-node and those constructed manually in the old way.
+
+  * Instead of waiting for the failure mode to materialize and then writing a patch in a rush, we could add an optional config var to op-node to switch it back into the old behavior. Rolling out the fix would not then require any software releases.
+
+  * op-geth could be modified to log a critical error (triggering an alert) if the withdrawals list in the body is ever non empty.
+
+- **Detection:** 
+  Fault proof monitoring systems may not detect this failure mode immediately, until an actor running patched software made a proposal or challenge.
+
+- **Recovery Path(s)**:
+  Fault proof infra would nee to be pointed at a patched op-node. The patch would restore the old behaviour for generating output roots.
+
+
+## Generic failure modes: 
+See the [generic FMA](./fma-generic-hardfork.md):
+* Chain halt at activation (there is a change to the engine API, which elevates this risk)
+* Activation failure
+* Invalid setImplementation execution
+* Chain split (across clients)
+
+## Specific Action Items
+- [ ] (BLOCKING) e2e tests must check for consistency between output roots returned from op-node and those constructed manually in the old way
+- [ ] (non-BLOCKING) op-node could be furnished with an override to make it serve output roots in the legacy fashion; this would also aid in testing (see above item). It would even allow us to run the two systems side by side for a time before fully switching over.
+- [ ] (non-BLOCKING) op-geth could be made to log a critical error triggering an alert if ever the `withdrawals` list in the block body is non empty (post Isthmus)
+
+## Generic Action Items
+- [x] (BLOCKING): We have implemented extensive unit and end-to-end testing of the activation flow: https://github.com/ethereum-optimism/optimism/blob/develop/op-e2e/actions/upgrades/isthmus_fork_test.go
+- [ ] (BLOCKING): We have implemented multi-client testing to reduce the chance of bugs (the above test could be migrated to a fault proof test where it can run on kona)
```

Comment:
Kurtosis / devnet multiclient testing might be a better / lower lift way of performing the multi client testing.

Reply:
We have this acceptance test /Users/georgeknee/code/ethereum-optimism/optimism/op-acceptance-tests/tests/isthmus/withdrawal_root_test.go

Reply:
a devnet-sdk test should suffice for this, which we already have one

Reply:
Do you mean this test https://github.com/ethereum-optimism/optimism/blob/develop/op-acceptance-tests/tests/isthmus/withdrawal_root_test.go ? I think it needs to be strengthened to make it a true multi-client test.